### PR TITLE
Js refactor

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Comment by grs** [comment]
Yes, the title of this PR was too general! I did not intend to imply it is the only refactoring needed. I agree that there is plenty more to do. This is really just the first step, to eliminate duplicated (and diverging) code.

**Comment by lulf** [comment]
Ok, thats fine. Was just wondering how one would go about doing that, and if its something that goes against JS convention :)

Merge at will!

**Comment by grs** [comment]
I think what is needed is more clarity on the abstractions relied on by different components, and then ensure translation as necessary on input/output from those. i.e. better defined (and better designed) interfaces between different pieces.

File: agent/lib/pod_watcher.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 Red Hat Inc.
+ *
```
Comment (rgodfrey) [inline agent/lib/pod_watcher.js]: (c) 2017 perhaps :)

File: agent/lib/topic.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016 Red Hat Inc.
```
Comment (rgodfrey) [inline agent/lib/topic.js]: 2017 :)

**Comment by lulf** [comment]
Merged manually via cli

**Pull request closed without merging**